### PR TITLE
Make Kotlin generators caching behaviour consistent with Gradle 8.8 

### PR DESCRIPTION
**Pull request by asodja**
Fixes https://github.com/gradle/gradle/issues/29458

Relates to https://github.com/gradle/gradle/pull/29534

### CODE REVIEW

**Comment by asodja** [comment]
@bot-gradle test this

File: platforms/core-execution/execution/src/main/java/org/gradle/internal/execution/UnitOfWork.java [hunk 1]
```diff
@@ -43,6 +44,8 @@
 
 public interface UnitOfWork extends Describable {
 
+    CachingDisabledReason NOT_WORTH_CACHING = new CachingDisabledReason(CachingDisabledReasonCategory.NOT_CACHEABLE, "Not worth caching.");
```
Comment (ov7a) [inline platforms/core-execution/execution/src/main/java/org/gradle/internal/execution/UnitOfWork.java:47]: 🤔 Should it be static?

Comment (lptr) [reply platforms/core-execution/execution/src/main/java/org/gradle/internal/execution/UnitOfWork.java:47]: It's in an interface, this is already static.

**Review by lptr — APPROVED** [review]
LGTM.

**Pull request merged**